screeps-game-api/src/objects/impls/construction_site.rs

```diff
@@ -3,17 +3,24 @@ use stdweb::unstable::TryInto;
 use {ReturnCode, StructureType};
 
 // TODO: Use root import after https://github.com/rust-lang/rust/issues/53140 is fixed.
-use super::super::ConstructionSite;
+use super::super::{ConstructionSite, HasId};
 
 simple_accessors! {
     ConstructionSite;
-    (id -> id -> String),
     (my -> my -> bool),
     (progress -> progress -> i32),
     (progress_total -> progressTotal -> i32),
     (structure_type -> structureType -> StructureType),
 }
 
+impl PartialEq for ConstructionSite {
```

Comment:
Do you think it would be at all reasonable to move automatic `PartialEq` / `Eq` implementations from `impl_structure` into an `impl_has_id` macro so that all things with ids could be covered?

I think things which implement `HasId` should be a strict superset of things implementing `StructureProperties`.

I guess it would mean we'd have to lay out every implementor which also implements from `StructureProperties` rather than having a blanket implementation, but it might be worth it to remove the need for duplicating these `PartialEq` implementations?

Reply:
Agreed